src/peep/peep.c

```diff
@@ -7772,13 +7772,404 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
+	// TODO: check to make sure all of the conditions are covered
+	// TODO: peep_has_food() is redundant with peep_get_standard/extra_item_index()
+	if ((shopItem >= 32 ? EXTRA_FOOD_MASK : STANDARD_FOOD_MASK) & (1 << shopItem2)) { // If the shop offers food
+		if (peep_has_food(peep)) { // If the peep already has food
+			int food = bitscanforward(peep->item_standard_flags & STANDARD_FOOD_MASK);
+			if (food != -1)
+				peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_HAVENT_FINISHED, food);
+			else
+				peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_HAVENT_FINISHED, (bitscanforward(peep->item_extra_flags & EXTRA_FOOD_MASK) + 32));
+			return 0;
+		}
+		else { // Peep does NOT have food
+			if (peep->nausea >= 145)
+				return 0;
+		}
+	}
+
+///loc_69AFD3:
+	// Don't buy certain items if it's raining?
+	if (!((shopItem >= 32 ? 0x80 : 0x00080301) & (1 << shopItem2))) {
+		if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_RAIN_LEVEL, sint8) > 0)
+			return 0;
+	}
+
+///loc_69B007:
+	// Don't buy certain items if it's too cold out?
+	if (!((shopItem >= 32 ? 0x80 : 0x100) & (1 << shopItem2))) {
+		if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_TEMPERATURE, sint8) < 12)
+			return 0;
+	}
+
+///loc_69B03B:
+	// See also: peep_has_drink()
+	// Checks to see if the peep is hungry enough to buy something
+	if (!((shopItem >= 32 ? 0x00118F48 : 0x012BA3C0) & (1 << shopItem2))) { // If the shop is selling a food item (NOT drinks)
+		if (peep->hunger < 75) {
+			peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_NOT_HUNGRY, 0xFF); // Null arg?
+			return 0;
+		}
+	}
+
+///loc_69B069:
+	// Checks to see if the peep is thirsty enough to buy something (similar to above)
+	if (!((shopItem >= 32 ? 0x00007030 : 0x02400020) & (1 << shopItem2))) {
+		if (peep->thirst < 75) {
+			peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_NOT_THIRSTY, 0xFF);
+			return 0;
+		}
+	}
+
+///loc_69B097:
+	//0x14001B = standard: balloon, toy, photo, umbrella, hat, t-shirt (but not map)
+	//0x87 = photo 2/3/4, sunglasses
+	if (shopItem == 4) { // umbrella?
+		if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_RAIN_LEVEL, sint8) != 0)
+			goto loc_69B119;
+	}
+
+///loc_69B0A4:
+	if (shopItem == 2) { // map?
+		if (peep->guest_heading_to_ride_id == 0xFF)
+			goto loc_69B119;
+	}
+
+///loc_69B0B1:
+	if ((!((shopItem >= 32 ? 0x87 : 0x14001B) & (1 << shopItem2))) && !RCT2_GLOBAL(0x00F1AEE3, unsigned char)) {
+		if (((scenario_rand() & 0x7F) + 0x73) < peep->happiness)
+			return 0;
+		else if (peep->no_of_rides < 3)
+			return 0;
+	}
+
+loc_69B119:
+	uint16 dx;
+	if (!RCT2_GLOBAL(0x00F1AEE3, unsigned char)) {
+		if (price != 0) {
+			if (peep->cash_in_pocket == 0) {
+				peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_SPENT_MONEY, 0xFF);
+				return 0;
+			}
+			if (price > peep->cash_in_pocket) {
+				peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_CANT_AFFORD, shopItem);
+				return 0;
+			}
+		}
+
+		///loc_69B143:
+		// There are six tangentally-related array accesses in this chunk of code.
```

Comment:
This is the price peeps are willing to pay depending on the weather. There will be a standard and extra flags version of each array and it looks like it has a >= 21 entry and a <=11 entry and one in between, the fourth entry appears to not be used anywhere.